libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1793,7 +1794,23 @@ void TypeChecker::expectType(Expression const& _expression, Type const& _expecte
 				_expectedType.toString() +
 				"."
 			);
-	}		
+	}
+
+	if (
+		type(_expression)->category() == Type::Category::RationalNumber &&
+		_expectedType.category() == Type::Category::FixedBytes
+	)
+	{
+		auto literal = dynamic_cast<Literal const*>(&_expression);
+
+		if (literal && !boost::starts_with(literal->value(), "0x"))
```

Comment:
Yes, I actually thought about that, should I use `looksLikeHex` or `isHexNumber`?